### PR TITLE
feat(fortran): Update queries

### PR DESCRIPTION
**Pull request by navdeeprana**
From https://github.com/stadelmanma/tree-sitter-fortran/pull/78

### CODE REVIEW

**Review by theHamsta — APPROVED** [review]

File: queries/fortran/highlights.scm [hunk 1]
```diff
@@ -5,147 +5,149 @@
...
+ (intrinsic_type)
+ "allocatable"
+ "attributes"
+ "device"
+ "dimension"
+ "endtype"
+ "global"
+ "grid_global"
+ "host"
+ "import"
```
Comment (theHamsta) [inline queries/fortran/highlights.scm:17]: I don't think all of these are `@type`

File: queries/fortran/highlights.scm [hunk 2]
```diff
@@ -157,26 +159,38 @@
...
 
 (end_subroutine_statement
   (name) @function)
 
+(end_module_procedure_statement
+  (name) @function)
+
 (subroutine_call
-  (identifier) @function.call)
+  (identifier) @function)
```
Comment (theHamsta) [inline queries/fortran/highlights.scm:194]: Why is this removed?

**Review by theHamsta — CHANGES_REQUESTED** [review]
Added some small comments

**Comment by ObserverOfTime** [comment]
The previous groups were better.

File: queries/fortran/highlights.scm (hunk 2 above)
Comment (navdeeprana) [reply queries/fortran/highlights.scm:194]: A mistake. I will fix it.

File: queries/fortran/highlights.scm (hunk 1 above)
Comment (navdeeprana) [reply queries/fortran/highlights.scm:17]: `import` goes in `@keyword`, I think. The rest are subroutine / function attributes [[1]](https://docs.nvidia.com/hpc-sdk/compilers/cuda-fortran-prog-guide/#cfpg-sub-func-qualifiers). For example
```fortran
attributes(host) function foo(x) result(y)
    real, intent(in) :: x
    real :: y
    y = x + 1
end function foo
```
 Should I move them to `@attribute` or `@keyword`?

Comment (theHamsta) [reply queries/fortran/highlights.scm:17]: Isn't `import` `@include`?

Comment (navdeeprana) [reply queries/fortran/highlights.scm]: moved `import` and `include` to `@include`. `global` etc to `@keyword`.

**Review by theHamsta — APPROVED** [review]

File: queries/fortran/highlights.scm [hunk 3]
```diff
@@ -5,147 +5,149 @@
...
+ "endtype"
+ "in"
+ "inout"
+ "intent"
+ "optional"
+ "out"
+ "pointer"
+ "type"
+ "value"
+ ] @type
```
Comment (theHamsta) [inline queries/fortran/highlights.scm:22]: I don't think type applies to any of the keywords

**Review by theHamsta — CHANGES_REQUESTED** [review]
@ObserverOfTime

File: queries/fortran/highlights.scm [hunk 4]
```diff
@@ -5,147 +5,149 @@
...
-] @type.definition
+ (none)
+ "implicit"
+ ] @attribute
 
 [
-  "in"
-  "inout"
-  "out"
-] @storageclass
```
Comment (theHamsta) [inline queries/fortran/highlights.scm:35]: they should stay `@storageclass`

**Review by ObserverOfTime — COMMENTED** [review]
Some keywords should be `@type.qualifier` (i.e. modifiers). I'll let you decide which.

**Comment by navdeeprana** [comment]
Thanks @amaanq for the better version in PR #4389. I guess this can be closed now?
